### ISSUE TITLE
Showing the line range that caused the error

### ISSUE THREAD

**Issue by elken** [adrienverge/yamllint#681]
Thanks for yamllint, this is a great tool!

The usecase we have for it is ensuring that our Helm templates don't include things like duplicate keys and such that `helm lint` doesn't pick up by default; and as such we pipe the result of the command that generates the output into yamllint.

Because there's no file that we can look at, having the failed line numbers isn't _quite_ that useful as it's built from multiple templates.

Would be really good if this could be added 😄 Thanks!

**Comment by adrienverge** [comment]
Hello ! I'm not sure I understand well what you mean. You don't care about the failed lines and columns, but you care about the subpart of the YAML code that caused the failure. Is that it?

**Comment by andrewimeson** [comment]
Perhaps a new output format (`--format`) called "contextual" that looks similar to ShellCheck's default output?

<img width="696" alt="image" src="https://github.com/user-attachments/assets/a6046a50-3c54-41af-84d9-c74d6c959101">

**Comment by elken** [comment]
Exactly that yes! I don't think it needs to be too fancy, just show the line range (maybe behind an option in case people want the current behaviour instead) and 1/2 extra either side for context.

**Comment by guitarrapc** [comment]
I'm doing similar with simple sed.

```shell
    if ! result=$(echo "$yaml" | yamllint --format "parsable" -); then
      # error message
      echo "$result"

      line=$(echo "$result" | cut -d':' -f2)
      context_lines=3  # line numer to show before&after (e.g. 1 line)

      # Adjust not to exceed under 1st line
      start_line=$((line - context_lines))
      if [ "$start_line" -lt 1 ]; then
        start_line=1
      fi
      # Adjust not to exceed under final line
      total_lines=$(echo "$manifest" | wc -l)
      end_line=$((line + context_lines))
      if [ "$end_line" -gt "$total_lines" ]; then
        end_line=$total_lines
      fi

      # show line
      echo "$manifest" | cat -n - | sed -n "${start_line},${end_line}p"
      exit 1
    fi
```
